src/net_processing.cpp

```diff
@@ -4268,8 +4295,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         if (received_new_header) {
```

Comment:
Since it's possible for `ProcessNewBlockHeaders` to return false without us returning, I wonder if we should only log if it returned `true` (as we do above).

Reply:
I don't think this is possible though?
The call structure is `ProcessNewBlockHeaders()` -> `AcceptBlockHeader()` -> {`CheckBlockHeader()` / `ContextualCheckBlockHeader()`}, and the latter two functions return either `true` or `state.Invalid(...)`. As a result, `ProcessNewBlockHeaders()` would always return the invalid `state` of the first header that fails, and  `net_processing` would abort.